### PR TITLE
Unify "draw per frame" and "draw per model" code

### PR DESCRIPTION
**Pull request by gyagp**
Previous code uses different functions for draw per frame and draw per
model path, which means many duplications. This patch will simplify
this.

### CODE REVIEW

File: src/aquarium-optimized/Aquarium.cpp [hunk 1]
```diff
@@ -780,46 +780,53 @@ void Aquarium::render() {
...
+    updateWorldMatrix(model);
   }
 }
 
-void Aquarium::drawBackground() {
-  Model *model = mAquariumModels[MODELNAME::MODELRUINCOlOMN];
-  for (int i = MODELNAME::MODELRUINCOlOMN; i <= MODELNAME::MODELSEAWEEDB; ++i) {
-    model = mAquariumModels[i];
-    model->draw();
+void Aquarium::updateWorldMatrix(Model *model) {
```
Comment (hujiajie) [inline src/aquarium-optimized/Aquarium.cpp:804]: Can we meld this into Aquarium::updateBackground()?

File: src/aquarium-optimized/Aquarium.cpp [hunk 2]
```diff
@@ -875,79 +882,22 @@ void Aquarium::updateAndDrawFishes() {
                static_cast<float>(M_PI) * 2),
           ii);
 
-      model->updatePerInstanceUniforms(worldUniforms);
-      model->draw();
+      if (mDrawPerModel) {
```
Comment (hujiajie) [inline src/aquarium-optimized/Aquarium.cpp:885]: This is a branch in the hottest path of Aquarium, but I don't know if that still matters in 2020.

File: src/aquarium-optimized/Aquarium.cpp [hunk 3]
```diff
@@ -780,46 +780,53 @@ void Aquarium::render() {
...
+  ASSERT(w.size() == 16);
+  memcpy(worldUniforms.world, w.data(), 16 * sizeof(float));
+  matrix::mulMatrixMatrix4(worldUniforms.worldViewProjection,
+                           worldUniforms.world,
+                           lightWorldPositionUniform.viewProjection);
+  matrix::inverse4(g.worldInverse, worldUniforms.world);
+  matrix::transpose4(worldUniforms.worldInverseTranspose, g.worldInverse);
+}
+
+void Aquarium::updateFishes() {
```
Comment (hujiajie) [inline src/aquarium-optimized/Aquarium.cpp:829]: The function name seems a little confusing now, as it means "update & draw" in some cases.

**Comment by gyagp** [comment]
Created a new one, so this one is closed.

**Pull request closed without merging**